### ISSUE TITLE
Daniel, please contact me

### ISSUE THREAD

**Issue by JeremyRand** [namecoin/meta#57]
@domob1812 I'm trying to email you but it looks like Vincent's mail server is still having issues.  Can you contact me via either Ricochet or Cwtch?  My usernames are in the namecoin.org repo, feel free to check OpenPGP sigs on the commit that added them.  Thanks!

**Comment by domob1812** [comment]
EDIT:  I'm taking this back.  Apparently my install of Ricochet (using the Debian Buster package) fails to store any state between starts (even though it created `~/.local/share/Ricochet`), and thus I lost you and the identity I posted.

Is there something I'm doing wrong in order to let it keep my data?

**Comment by JeremyRand** [comment]
@domob1812 IIRC there's a bug in the Debian package for Ricochet.  Using the upstream release (either their binaries or building from upstream source) should work.

EDIT: Or if you have easy access to other distros, I believe Fedora's package works fine.

**Comment by domob1812** [comment]
Ok thanks, I'll try using the source itself then.  If I manage to get it set up, I'll post my ID here with OpenPGP signature and add you.

**Comment by domob1812** [comment]
@JeremyRand I've now built Ricochet from source, but am getting the same issue.  It seems their repository has not been updated since 2016 anyway?

**Comment by JeremyRand** [comment]
@domob1812 Looks like the Debian package bug is specifically in the AppArmor policy.  Look up issue 594 in the upstream repo; replace the `owner @{HOME}` lines at the bottom of the policy that you have with the (longer) section from that issue, then reload the AppArmor service via `systemctl`.  That fixed the Debian package for me.  (I have no idea why Debian's maintainer didn't add this fix, maybe I'll pester them about it....)

**Comment by JeremyRand** [comment]
@domob1812 More specifically, look at the bottom of `/etc/apparmor.d/usr.bin.ricochet` for a set of rules that relate to `owner @{HOME}`.  For me on Debian Buster, there were 2 such lines, both of which related to `owner @{HOME}/.local/`.  Replace those lines with the following:

~~~
  owner @{HOME}/.local/share/Ricochet/ rwl,
  owner @{HOME}/.local/share/Ricochet/** mrwkl,

  owner @{HOME}/.cache/Ricochet/ rw,
  owner @{HOME}/.cache/Ricochet/** rwkl,

  owner @{HOME}/.config/gtk-3.0/* r,
~~~

Source: https://github.com/ricochet-im/ricochet/issues/594

This fixed the Debian-packaged's Ricochet's ability to preserve state between runs for me.

**Comment by domob1812** [comment]
Nope, no luck with that either.  If I edit the apparmor profile like that, my Ricochet still does not keep any state.

If it is important to get in touch, can you use Bitmessage and/or XMPP (with OTR) instead?

**Comment by domob1812** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

It seems I got it working now. My ID is:

ricochet:guyb567vuaqxgk5z
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEEYvsMrbXPhv14+R958cnIcJ5uTjoFAl6X3UkACgkQ8cnIcJ5u
TjrWDBAAo7bB3EBP2d2Q4Lo5KHmi0n1qP4vzXS62T2EHsw8kI/F1BZnJhvnGwF5V
HqIMMYN1RewGeAbw/N1AB5/zEkLtPr0i21FVs7pdLtOvxUf8ghond7MN8+S7Rdoq
y6va1YzERf024qt6PMdiiy2+1WKK6ljzJoz2zlgTQMZhxNBmy8103sCjJSfJPaUC
xT95ZvdaXiPIFeG1vbnFq0iv7tF6ACOZWlYqH2QOd2pYNLpEgDOaTL8xsaDWid7q
E7sigqnAnjASnKp5ujUpzeqO9Zzr+leZjpp95BuSJ7gCiXSdTL2NfVd1CezXGlWs
ucTSNmatLjjrc3OvSCIpe0+LzVPTes2rYET/PXjB2BoIv5GYyprkMhQ+Z7EUX/Tx
DraPnluyNgTyedVkpDO7+ix2wWsi2fCP+Fpw0HFzNoiZWmeEkwj7Fn2n91/CvUNL
GgzAecL8MMIsYPbDg1DjA1JTGEa6b+ahZeTuCwvu7mqPNChCeoeKFspnK1llPmCJ
pAsx91BPGZH8XdLz5IrkHKfz138kAu0HBGNR1cUMobdVE8cxQHPiDC9o28AKGhh4
rzjQhOoBDame9GLdXP4SqKHIj0NNDNXO/+vCCqNmO2Q6rHI9xz9sI1Z7ZGRNREWU
+mLqaIxJUrnkn0fsL727K5IDVlvV1/hW0btxqPjxhNPSBiEMwso=
=FxJC
-----END PGP SIGNATURE-----

EDIT:  It seems Github messes up whitespace, so I had to give it a couple of tries to make my signature validate.  This one should now.
